huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go

```diff
@@ -0,0 +1,376 @@
+									},
+									"specification": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: `The compute resource type. Currently, resource types A, B, and C are available.`,
+									},
+									"resources": {
+										Type:        schema.TypeList,
+										Computed:    true,
+										Description: `The list of resource objects.`,
+										Elem:        temBodResElem(),
+									},
+									"driver_cores": {
+										Type:        schema.TypeInt,
+										Computed:    true,
+										Description: `The number of CPU cores of the Spark application driver.`,
+									},
+									"num_executors": {
+										Type:        schema.TypeInt,
+										Computed:    true,
+										Description: `The number of Executors in a Spark application.`,
+									},
+								},
+							},
+						},
+						"group": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The group of template.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// temBodResElem
+// The Elem of "templates.body.resources"
+func temBodResElem() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The name of resource.`,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The type of resource.`,
+			},
+		},
+	}
+}
+
+// temBodDepPacElem
+// The Elem of "templates.body.dependent_packages"
+func temBodDepPacElem() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The name of a user group.`,
+			},
+			"resources": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: `The resources of a user group.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The name of resource.`,
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The type of resource.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+type SparkTemplatesDSWrapper struct {
+	*schemas.ResourceDataWrapper
+	Config *config.Config
+}
+
+func newSparkTemplatesDSWrapper(d *schema.ResourceData, meta interface{}) *SparkTemplatesDSWrapper {
+	return &SparkTemplatesDSWrapper{
+		ResourceDataWrapper: schemas.NewSchemaWrapper(d),
+		Config:              meta.(*config.Config),
+	}
+}
+
+func dataSourceDliSparkTemplatesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	wrapper := newSparkTemplatesDSWrapper(d, meta)
+	lisSpaJobTemRst, err := wrapper.ListSparkJobTemplates()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, _ := uuid.GenerateUUID()
```

Comment:
The error message returned by the function cannot be listed randomly.

Reply:
done